src/components/EditTodoForm/EditTodoForm.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+
+interface Props {
+  onTodoChanged: (newTitle: string) => void,
+  onCanceled: () => void,
+  initialTitle: string,
+}
+
+export const EditTodoForm: React.FC<Props> = ({
+  onTodoChanged,
+  onCanceled,
+  initialTitle,
+}) => {
+  const [newTitle, setNewTitle] = useState(initialTitle);
+
+  const handleBlur = () => {
+    onTodoChanged(newTitle);
+  };
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    onTodoChanged(newTitle);
+  };
+
+  const keyboardListener = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      onCanceled();
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('keyup', keyboardListener);
+
+    return () => window.removeEventListener('keyup', keyboardListener);
+  }, []);
+
+  return (
+    <form onSubmit={onSubmit}>
+      <input
+        // eslint-disable-next-line jsx-a11y/no-autofocus
+        autoFocus
```

Comment:
I'm not sure, but as I think, it's a bad practice to set focus by this way. Try to use some react methods to do that. (useRef, someRef.current.focus()).

Reply:
it is right ( but it is not critical)